docs/contracts/visibility-and-getters.rst

```diff
@@ -157,12 +157,21 @@ it evaluates to a state variable.  If it is accessed externally
         }
     }
 
+Getter functions do not always return the entire state variable,
+arrays and structs are two exceptions.
+
 If you have a ``public`` state variable of array type, then you can only retrieve
 single elements of the array via the generated getter function. This mechanism
-exists to avoid high gas costs when returning an entire array. You can use
-arguments to specify which individual element to return, for example
-``myArray(0)``. If you want to return an entire array in one call, then you need
-to write a function, for example:
+exists to avoid high gas costs when returning an entire array.
+When you declare a public state variable of a struct type,
+the generated getter function does not return the struct as a single object in memory.
+Instead, due to historical design choices and early limitations in the ABI encoding of structs,
+the getter function returns the individual members of the struct as separate elements within a tuple.
+If the struct contains multiple members, the getter function returns a tuple with each member as a
+distinct element. Conversely, if the struct has only one member, the getter function simply returns
+that member directly instead of wrapping it in a tuple.
```

Comment:
```suggestion
If the struct has only one member, the getter function returns that member's value.
```

The removed phrase was redundant.

Also, I think that technically it still returns a unitary tuple, which in the end is evaluated/considered the same as the sole content of the tuple.

Reply:
Also, the text here is not entirely correct. There are cases where the struct members can be omitted.